### PR TITLE
make Policy init via duck typing

### PR DESCRIPTION
**Pull request by n8pease**
removes kwarg style input arguments

### CODE REVIEW

File: tests/testPolicy.py [hunk 1]
```diff
@@ -111,19 +111,23 @@ def testMerge(self):
...
-        loadedPolicy = Policy(filePath='testDumpFile.yaml')
+        loadedPolicy = Policy('testDumpFile.yaml')
         self.assertEqual(self.policy, loadedPolicy)
         os.remove('testDumpFile.yaml')
 
     def testNonExistantPolicyAtPath(self):
-        self.assertRaises(IOError, Policy, filePath="c:/policy.yaml")
-        self.assertRaises(IOError, Policy, filePath="c:/policy.paf")
-        self.assertRaises(IOError, Policy, filePath="c:/policy")
+        self.assertRaises(IOError, Policy, "c:/policy.yaml")
```
Comment (timj) [inline tests/testPolicy.py]: Can you rewrite these using `with` so it's clearer what the test is.

Comment (timj) [inline tests/testPolicy.py]: It's a shame `FileNotFoundError` seems to be Python 3 only.

File: python/lsst/daf/persistence/policy.py [hunk 2]
```diff
@@ -250,6 +210,24 @@ def __contains__(self, key):
...
+    def defaultPolicyFile(productName, filePath, repos=None):
+        """Get the path to a default policy file
+
+        @param productName (string) The name of the product that the default policy is installed as part of
+        @param filepath (string) The relative pathname to the policy file.  
+        @param repos (string) the subdirectory with the product's install directory where policy files are 
+                              stored. If None (default), the filepath argument is relative to the installation
+                              directory.
+        """
+        basePath = os.getenv(productName.upper() + "_DIR")
```
Comment (timj) [inline python/lsst/daf/persistence/policy.py]: Why use this instead of the `getPackageDir` that was in place before?

File: python/lsst/daf/persistence/policy.py [hunk 3]
```diff
@@ -69,25 +69,36 @@ class Policy(_PolicyBase):
...
+    
         """
         collections.UserDict.__init__(self)
-        self.__init(other, preference, **kwargs)
+        
+        if other is None:
+            return
+
+        try:
+            other.keys() # is it a dict?
```
Comment (timj) [inline python/lsst/daf/persistence/policy.py]: The spacing to the comment is wrong here. Did you check the code was compliant (with flake8)?

Comment (timj) [inline python/lsst/daf/persistence/policy.py]: Also, why not just ask if it's dict-like and fit it in with the rest of the tests below?

File: python/lsst/daf/persistence/policy.py [hunk 4]
```diff
@@ -250,6 +210,24 @@ def __contains__(self, key):
                 return False
         return keys[-1] in d
 
+    @staticmethod
+    def defaultPolicyFile(productName, filePath, repos=None):
+        """Get the path to a default policy file
+
+        @param productName (string) The name of the product that the default policy is installed as part of
+        @param filepath (string) The relative pathname to the policy file.
```
Comment (timj) [inline python/lsst/daf/persistence/policy.py]: "The location of the file relative to the product root directory."?

File: python/lsst/daf/persistence/policy.py [hunk 5]
```diff
@@ -250,6 +210,24 @@ def __contains__(self, key):
...
 
+    @staticmethod
+    def defaultPolicyFile(productName, filePath, repos=None):
+        """Get the path to a default policy file
+
+        @param productName (string) The name of the product that the default policy is installed as part of
+        @param filepath (string) The relative pathname to the policy file.  
+        @param repos (string) the subdirectory with the product's install directory where policy files are 
+                              stored. If None (default), the filepath argument is relative to the installation
+                              directory.
```
Comment (timj) [inline python/lsst/daf/persistence/policy.py]: Ok, I'm actually having a hard time working out how `filepath` and `repos` relate to each other from this description without looking at the code. Is the actual file in `productDir/repos/filepath`? Is it also `productDir/filepath` if `repos` is `None`? Does `filepath` include the actual file itself or just the path to the file? If it includes the file does it include the extension? All these questions were in my head as I read this description.

File: python/lsst/daf/persistence/policy.py (hunk 2 above)
Comment (n8pease) [inline python/lsst/daf/persistence/policy.py]: this got cut & pasted from a code I wrote before I knew about getPackageDir. will fix.

File: python/lsst/daf/persistence/policy.py (hunk 5 above)
Comment (n8pease) [inline python/lsst/daf/persistence/policy.py]: This reimplements (in python) code from pex_policy. The documentation is was moved from there. You're right though, it's hard to parse. I'll clean it up.

File: tests/testPolicy.py [hunk 6]
```diff
@@ -111,19 +111,25 @@ def testMerge(self):
...
+        loadedPolicy = Policy('testDumpFile.yaml')
         self.assertEqual(self.policy, loadedPolicy)
         os.remove('testDumpFile.yaml')
 
     def testNonExistantPolicyAtPath(self):
-        self.assertRaises(IOError, Policy, filePath="c:/policy.yaml")
-        self.assertRaises(IOError, Policy, filePath="c:/policy.paf")
-        self.assertRaises(IOError, Policy, filePath="c:/policy")
+        # Test that loading paf & yaml files that do not exist raises an IOError.
+        self.assertRaises(IOError, Policy, "c:/nonExistantPath/policy.yaml")
```
Comment (timj) [inline tests/testPolicy.py]: Please use
```python
with self.assertRaises(IOError):
    Policy("c:....yaml")
```
as it is much clearer how the command is constructed that triggers the exception.
